common/common.h

```diff
@@ -141,6 +141,9 @@ struct gpt_params {
     bool use_color         = false; // use color to distinguish generations and inputs
     bool interactive       = false; // interactive mode
     bool chatml            = false; // chatml mode (used for models trained on chatml syntax)
```

Comment:
Is it time to remove the `chatml` parameter? Seems it is superseded by this PR

Reply:
If everyone is ok with the new flow from this PR, then we can potentially eliminate the chatml mode and just retain the interactive mode and inturn this simplification built on top of the same wrt multiple models/standards, by using interactive modes in-prefix/suffix and antiprompt.